### PR TITLE
fix(history): fix double-prefixing in StackContext.Hash

### LINKED ISSUE #966: fix(history): fix double-prefixing in StackContext.Hash
## Overview

`StackContext.Hash()` in `internal/history/hash.go` (lines ~21-28) expands a short stack name to `org/project/stack`, then unconditionally rebuilds the string as `org/project/{expanded}`, producing `org/project/org/project/stack` — a double-prefixed hash input.

## User Story

As a FinFocus user with short stack names,
I want consistent hash computation,
So that the same logical stack always maps to the same history bucket regardless of stack name format.

## Problem Statement

**Current code:**
```go
func (sc StackContext) Hash() string {
    stack := sc.Stack
    if stack != "" && !strings.Contains(stack, "/") {
        stack = fmt.Sprintf("%s/%s/%s", sc.Organization, sc.Project, stack) // expand short name
    }
    s := fmt.Sprintf("%s/%s/%s", sc.Organization, sc.Project, stack) // ALWAYS prefixes again
    h := sha256.Sum256([]byte(s))
    return hex.EncodeToString(h[:])[0:16]
}
```

**Scenario:**
- Input: `StackContext{Organization: "org", Project: "proj", Stack: "dev"}`
- Line 24: `stack` becomes `"org/proj/dev"`
- Line 26: `s` becomes `"org/proj/org/proj/dev"` — double-prefixed
- Already-qualified stacks (`"org/proj/dev"`) would produce `"org/proj/org/proj/dev"` too

**Impact:** Different logical representations of the same stack produce different hashes, causing history fragmentation.

### Technical Approach

Compute one canonical stack string and hash that directly:
```go
func (sc StackContext) Hash() string {
    canonicalStack := sc.Stack
    if canonicalStack != "" && !strings.Contains(canonicalStack, "/") {
        canonicalStack = fmt.Sprintf("%s/%s/%s", sc.Organization, sc.Project, canonicalStack)
    }
    h := sha256.Sum256([]byte(canonicalStack))
    return hex.EncodeToString(h[:])[0:16]
}
```

Or if the intent is to always include org/project:
```go
s := fmt.Sprintf("%s/%s/%s", sc.Organization, sc.Project, sc.Stack)
```

### Files Likely Affected

- `internal/history/hash.go` — `StackContext.Hash()` method

## Acceptance Criteria

- [ ] Short stack names ("dev") and fully-qualified names ("org/proj/dev") produce the same hash
- [ ] No double-prefixing occurs
- [ ] Empty stack contexts produce a consistent (if meaningless) hash
- [ ] `make test` and `make lint` pass

## ⚠️ Migration Note

Fixing this changes existing hashes. Existing history data keyed by the old (double-prefixed) hash will not be found under the new hash. Consider whether a migration or fallback lookup is needed.

## Testing Strategy

- [ ] Unit test: short stack name produces same hash as fully-qualified
- [ ] Unit test: already-qualified stack name is not double-prefixed
- [ ] Unit test: empty fields produce consistent result

## Related

- Related to detectHistoryStackContext returning empty context issue
- Part of resource history store feature (branch 608-resource-history-store)

### PR DESCRIPTION
**Pull request by rshade**
## Summary

- Short stack names like `"dev"` were canonicalized to `"org/proj/dev"` but then re-prefixed to `"org/proj/org/proj/dev"` before hashing, causing short and fully-qualified names to produce different hashes for the same logical stack
- Replaced the redundant `fmt.Sprintf` with direct hashing of the canonical form so both `"dev"` and `"org/proj/dev"` now produce identical hashes
- Added comprehensive unit tests covering short/qualified equivalence, empty stack, zero-value context, determinism, and the `URNHash`/`BuildHistoryKey`/`BuildTagKey` helpers

## Test plan

- [x] `TestStackContextHash_NoDoublePrefixing` verifies hash matches `sha256("org/proj/dev")`, not the double-prefixed form
- [x] `TestStackContextHash_ShortAndQualifiedMatch` confirms short and fully-qualified names produce the same hash
- [x] Edge cases: empty stack, zero-value context, no-org CLI path
- [x] `make test` passes
- [x] `make lint` passes

### Modified files

- `internal/history/hash.go` - Removed redundant `fmt.Sprintf("%s/%s/%s", ...)` that re-prefixed the already canonical stack name; hash now computed directly from `canonical`

### New files

- `internal/history/hash_test.go` - Unit tests for `StackContext.Hash`, `URNHash`, `BuildHistoryKey`, and `BuildTagKey`

Closes #966

## Summary by CodeRabbit

* **Chores**
  * Improved internal hashing logic to ensure consistent, canonical hashing of stack identifiers without changing external behavior.

* **Tests**
  * Added thorough tests verifying deterministic, stable 16-character hash outputs, canonical equivalence for stack names, prevention of double-prefixing, and exact formatting of history and tag keys.

### CODE REVIEW

File: internal/history/hash.go [hunk 1]
```diff
@@ -19,12 +19,11 @@ type StackContext struct {
...
-	if stack != "" && !strings.Contains(stack, "/") {
-		stack = fmt.Sprintf("%s/%s/%s", sc.Organization, sc.Project, stack)
+	canonical := sc.Stack
+	if canonical != "" && !strings.Contains(canonical, "/") {
+		canonical = fmt.Sprintf("%s/%s/%s", sc.Organization, sc.Project, canonical)
 	}
-	s := fmt.Sprintf("%s/%s/%s", sc.Organization, sc.Project, stack)
-	h := sha256.Sum256([]byte(s))
+	h := sha256.Sum256([]byte(canonical))
 	return hex.EncodeToString(h[:])[0:16]
```
[the remark this answers is not in the document]
Comment (rshade) [reply internal/history/hash.go]: @coderabbitai  this hasn't been released yet, no need to be backward compatible
